src/tests/ftest/tags.py

```
@@ -650,84 +726,206 @@ def print_verbose(*args):
+
+    else:
+        width = []
+        for line in output:
+            for index, column in enumerate(line):
+                size = len(str(column))
+                if index == len(width):
+                    width.append(size)
+                elif size > width[index]:
+                    width[index] = size
+        format_line = "  ".join(list(map(lambda x: f"{{:{x}}}", width)))
+        for line in output:
+            print(format_line.format(*list(map(lambda x: str(x), line))))
+
+    return 0 if output else 1
+
+
+def __comma_separated_arg(val):
+    """Parse a comma-separated argument.
+
+    Args:
+        val (str): string to parse comma-separated values from
+
+    Returns:
+        set: comma-separated strings converted to a set
 
     Raises:
         ArgumentTypeError: if val is invalid
     """
     if not val:
-        raise ArgumentTypeError("tags cannot be empty")
+        raise ArgumentTypeError("comma-separated argument cannot be empty")
     try:
         return set(map(str.strip, val.split(",")))
     except Exception as err:  # pylint: disable=broad-except
-        raise ArgumentTypeError(f"Invalid tags: {val}") from err
+        raise ArgumentTypeError(f"Invalid comma-separated argument: {val}") from err
 
 
 def main():
-    """main function execution"""
-    description = '\n'.join([
-        'Commands',
-        '  lint - lint ftest avocado tags',
-        '  list - list ftest avocado tags associated with test files',
-        '  dump - dump the file/class/method/tag structure for test files',
-        '  unit - run self unit tests'
-    ])
-
-    parser = ArgumentParser(formatter_class=RawDescriptionHelpFormatter, description=description)
-    parser.add_argument(
-        "command",
-        choices=("lint", "list", "dump", "unit"),
-        help="command to run")
-    parser.add_argument(
+    """Main function execution.
+
+    Returns:
+        int: 0 = success
+             1 = error
+             2 = code problem
```

I don't think we should return rc 2. If there is a code issue we should just raise an exception.

---

Add exception in place of rc=2.